src/test/java/cn/wowspeeder/ss/SocksCommonUtilsTest.java

```diff
@@ -0,0 +1,71 @@
+package cn.wowspeeder.ss;
+
+import static org.mockito.AdditionalMatchers.or;
+import static org.mockito.Matchers.anyInt;
+import static org.mockito.Matchers.isA;
+import static org.mockito.Matchers.isNull;
+
+import com.diffblue.deeptestutils.mock.DTUMemberMatcher;
+import io.netty.buffer.ByteBuf;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+
+import java.lang.reflect.Method;
+import java.nio.charset.Charset;
+
+public class SocksCommonUtilsTest {
+
+  @Rule public final ExpectedException thrown = ExpectedException.none();
+
+  @Test
+  public void testIntToIp() {
+    Assert.assertEquals("11.8.0.8", SocksCommonUtils.intToIp(185_073_672));
+  }
+
+  @Test
+  public void testIpv6toCompressedForm() {
+    Assert.assertEquals("102:408:1020:407f:fffe:fcf0:e0c0:8000",
+            SocksCommonUtils.ipv6toCompressedForm(new byte[]{1, 2, 4, 8, 16,
+                    32, 64, 127, -1, -2, -4, -16, -32, -64, -128, 0}));
+  }
+
+  @Test
+  public void testIpv6toCompressedFormCompressionApplied() {
+    Assert.assertEquals("0::2:408:1020",
+            SocksCommonUtils.ipv6toCompressedForm(new byte[]{0, 0, 0, 0, 0, 0,
+                    0, 0, 0, 0, 0, 2, 4, 8, 16, 32}));
+  }
+
+  @Test
+  public void testIpv6toStr() {
+    Assert.assertEquals("0:0:0:0:0:2:408:1020",
+            SocksCommonUtils.ipv6toStr(new byte[]{0, 0, 0, 0, 0, 0, 0, 0, 0,
+                    0, 0, 2, 4, 8, 16, 32}));
+  }
+
+  @PrepareForTest(ByteBuf.class)
+  @Test
+  public void testResadUsAscii()
+      throws Exception {
+
+    final ByteBuf buffer = PowerMockito.mock(ByteBuf.class);
```

Comment:
`ByteBuf` is mocked because it is abstract - instead of mocking it you could just construct one using `Unpooled` - something like `Unpooled.copiedBuffer(new byte[]{'f', 'o', 'o'})`. Then you won't have to use `deeptestutils` and you can remove it as a dependency.

Reply:
Thanks i wanted to do this but just didn't know a way of doing so